src/dxil/wrapper.rs

```diff
@@ -0,0 +1,480 @@
+        let mut desc = d3d12shader::D3D12_FUNCTION_DESC::default();
+        return_hr_wrapped!(unsafe { self.inner.get_desc(&mut desc as *mut _) }, desc);
+    }
+
+    pub fn get_function_parameter(&self, parameter_index: i32) -> D3D12FunctionParameterReflection {
+        unsafe {
+            D3D12FunctionParameterReflection::new(
+                self.inner.get_function_parameter(parameter_index),
+            )
+        }
+    }
+
+    pub fn get_resource_binding_desc(
+        &self,
+        resource_index: u32,
+    ) -> Result<d3d12shader::D3D12_SHADER_INPUT_BIND_DESC, HassleError> {
+        let mut desc = d3d12shader::D3D12_SHADER_INPUT_BIND_DESC::default();
+        return_hr_wrapped!(
+            unsafe {
+                self.inner
+                    .get_resource_binding_desc(resource_index, &mut desc as *mut _)
+            },
+            desc
+        );
+    }
+
+    pub fn get_resource_binding_desc_by_name(
+        &self,
+        name: &CStr,
+    ) -> Result<d3d12shader::D3D12_SHADER_INPUT_BIND_DESC, HassleError> {
+        let mut desc = d3d12shader::D3D12_SHADER_INPUT_BIND_DESC::default();
+        return_hr_wrapped!(
+            unsafe {
+                self.inner.get_resource_binding_desc_by_name(
+                    name.as_ptr() as *const i8,
+                    &mut desc as *mut _,
+                )
+            },
+            desc
+        );
+    }
+
+    pub fn get_variable_by_name(&self, name: &CStr) -> D3D12ShaderReflectionVariable {
+        unsafe {
+            D3D12ShaderReflectionVariable::new(
+                self.inner.get_variable_by_name(name.as_ptr() as *const i8),
+            )
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct D3D12ShaderReflectionType {
+    inner: ComPtr<ID3D12ShaderReflectionType>,
+}
+
+impl D3D12ShaderReflectionType {
+    fn new(inner: ComPtr<ID3D12ShaderReflectionType>) -> Self {
+        Self { inner }
+    }
+
+    pub fn get_base_class(&self) -> D3D12ShaderReflectionType {
+        unsafe { D3D12ShaderReflectionType::new(self.inner.get_base_class()) }
+    }
+
+    pub fn get_desc(&self) -> Result<d3d12shader::D3D12_SHADER_TYPE_DESC, HassleError> {
+        let mut desc = d3d12shader::D3D12_SHADER_TYPE_DESC::default();
+        return_hr_wrapped!(unsafe { self.inner.get_desc(&mut desc as *mut _) }, desc);
+    }
+
+    pub fn get_inferface_by_index(&self, index: u32) -> D3D12ShaderReflectionType {
+        unsafe { D3D12ShaderReflectionType::new(self.inner.get_interface_by_index(index)) }
+    }
+
+    pub fn get_member_type_by_index(&self, index: u32) -> D3D12ShaderReflectionType {
+        unsafe { D3D12ShaderReflectionType::new(self.inner.get_member_type_by_index(index)) }
+    }
+
+    pub fn get_member_type_by_name(&self, name: &CStr) -> D3D12ShaderReflectionType {
+        unsafe {
+            D3D12ShaderReflectionType::new(
+                self.inner
+                    .get_member_type_by_name(name.as_ptr() as *const i8),
```

Comment:
Maybe we should stop making a mess and force `CHAR` (and everything based on it`) to be `char` instead of `i8`, as this type is platform dependent.